File: src/#429/solution.js
function Node(val, children) {
  this.val = val;
  this.children = children;
}

var levelOrder = function (root) {
  if (root === null) {
    return [];
  }
  let res = [];
  helper(root, 0);

  return res;
  function helper(root, level) {
    if (root === null) {
      return;
    }
    if (level >= res.length) {
      res.push([root.val]);
    } else {
      res[level].push(root.val);
    }
    root.children.forEach((node) => {
      helper(node, level + 1);
    });
  }
};

File: src/#429/solution.go
package solution

type Node struct {
	Val      int
	Children []*Node
}

func levelOrder(root *Node) [][]int {
	if root == nil {
		return [][]int{}
	}
	res := [][]int{}
	helper(root,0,&res)
	return res
}

func helper(root *Node, level int, res *[][]int) {
	if root == nil {
		return
	}

	if level >= len(*res) {
		*res = append(*res, []int{root.Val})
	} else {
		(*res)[level] = append((*res)[level], root.Val)
	}

	for _, node := range root.Children {
		helper(node, level+1, res)
	}
}
